api/v1alpha3/machinehealthcheck_webhook.go

```diff
@@ -119,7 +120,7 @@ func (m *MachineHealthCheck) validate(old *MachineHealthCheck) error {
 		)
 	}
 
-	if m.Spec.NodeStartupTimeout != nil && m.Spec.NodeStartupTimeout.Seconds() < 30 {
+	if m.Spec.NodeStartupTimeout != nil && m.Spec.NodeStartupTimeout.Seconds() < 30 && !disableMinimumNodeStartupTimeoutForTest {
```

Comment:
This was my idea, so can't comment. @vincepri may have opinions?

Reply:
What's the node startup timeout set to for tests?

Reply:
I see it now, it's 1ms, can we instead have a variable for the right side of `m.Spec.NodeStartupTimeout.Seconds() < 30`?

If we define a variable in this file like `minNodeStartupTimeout := 30 * time.Second`, we can create (if there isn't one already) a `machinehealthcheck_webhook_test.go` with

```go
func init() {
	minNodeStartupTimeout = 1 * time.Millisecond
}
```

The init of the test file will only run when tests are run, because test files have an implicit "test" build tag.

Reply:
the defaultNodeStartupTimeout is set to 10 seconds

Reply:
It looks like 
func init() {
	minNodeStartupTimeout = 1 * time.Millisecond
}
is not run when tests are run. in fact, minNodeStartupTimeout still has 30 Second as value, and hence m.Spec.NodeStartupTimeout.Seconds() < minNodeStartupTimeout is still failing...